src/test/java/AnimalTest.java

```diff
@@ -0,0 +1,68 @@
+import agh.ics.oop.*;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class AnimalTest {
+
+    @Test
+    public void AnimalTest1() {
```

Comment:
unikałbym takich nazw testów, fajnie żeby zawsze nazwa opisywała dany przypadek (a jeśli przypadki są ciężkie do nazwania to też sygnał, że może się dublują albo nie mają sensu ;p).

Btw, tu sam kod przypadków jest bardzo podobny, jakbyś chciał to uogólnić to polecam mechanizm `@ParametrizedTest`, na przykład w wersji CSV: https://www.baeldung.com/parameterized-tests-junit-5#4-csv-literals